ProcessLib/PhaseField/PhaseFieldFEM.h

```diff
@@ -107,6 +107,14 @@ template <typename ShapeFunction, typename IntegrationMethod,
           int DisplacementDim>
 class PhaseFieldLocalAssembler : public PhaseFieldLocalAssemblerInterface
 {
+private:
+    static constexpr int displacement_index = 0;
+    static constexpr int displacement_size =
+        ShapeFunction::NPOINTS * DisplacementDim;
+    static constexpr int phasefield_index =
+        displacement_index + displacement_size;
+    static constexpr int phasefield_size = ShapeFunction::NPOINTS;
+
```

Comment:
Are there any technical reasons to move these private internal definitions to the top of the class declaration before the public part? As a user I first want to see the interface and then maybe I'm interested in the private details.

Reply:
Yes, technical reason. The values are needed for typedefs below...